packages/jira-adapter/src/filters/add_references_to_project_schemes.ts

```diff
@@ -0,0 +1,171 @@
+}
+
+type FieldConfigurationSchemeProjects = {
+  fieldConfigurationScheme?: FieldConfigurationSchemeValue
+  projectIds: string[]
+}
+
+type IssueTypeScreenSchemeValue = {
+  id: string
+}
+
+type IssueTypeScreenSchemeInstance = InstanceElement & {
+  value: InstanceElement['value'] & IssueTypeScreenSchemeValue
+}
+
+type IssueTypeScreenSchemeProjects = {
+  issueTypeScreenScheme: IssueTypeScreenSchemeValue
+}
+
+export type ProjectInstance = InstanceElement & {
+  value: InstanceElement['value'] & {
+    workflowScheme: {workflowScheme: ReferenceExpression}[]
+    permissionScheme: ReferenceExpression[]
+    notificationScheme: ReferenceExpression[]
+    fieldConfigurationScheme: FieldConfigurationSchemeProjects[]
+    issueTypeScreenScheme: IssueTypeScreenSchemeProjects[]
+  }
+}
+
+type SchemeFieldsValues = {
+  workflowScheme: ReferenceExpression
+  permissionScheme: ReferenceExpression
+  notificationScheme: ReferenceExpression
+  issueTypeScreenScheme: ReferenceExpression | IssueTypeScreenSchemeValue
+  fieldConfigurationScheme: ReferenceExpression | FieldConfigurationSchemeValue | string
+}
+
+
+export const isProjectInstance = (element: InstanceElement): element is ProjectInstance => {
+  const isArrayWithSingleReference = (value: unknown): boolean =>
+    _.isArray(value)
+    && value.length === 1
+    && isReferenceExpression(value[0])
+  const hasIssueTypeScreenSchemeField = (): boolean => {
+    const isIssueTypeScreenSchemeProjects = (value: Value): boolean =>
+      _.isString(value.issueTypeScreenScheme?.id)
+    const issueTypeScreenSchemeField = element.value.issueTypeScreenScheme
+    return _.isArray(issueTypeScreenSchemeField)
+        && issueTypeScreenSchemeField.every(isIssueTypeScreenSchemeProjects)
+  }
+  const hasFieldConfigurationSchemeField = (): boolean => {
+    const isFieldConfigurationSchemeProjects = (value: Value): boolean =>
+      _.isArray(value.projectIds) && value.projectIds.every(_.isString)
+    const fieldConfigurationSchemeField = element.value.fieldConfigurationScheme
+    return _.isArray(fieldConfigurationSchemeField)
+        && fieldConfigurationSchemeField.every(isFieldConfigurationSchemeProjects)
+  }
+
+  return element.elemID.typeName === 'Project'
+      && hasIssueTypeScreenSchemeField()
+      && hasFieldConfigurationSchemeField()
+      && isReferenceExpression(element.value.workflowScheme[0]?.workflowScheme)
+      && isArrayWithSingleReference(element.value.permissionScheme)
+      && isArrayWithSingleReference(element.value.notificationScheme)
+}
+const isFieldConfigurationSchemeInstance = (element: InstanceElement)
+  : element is FieldConfigurationSchemeInstance =>
+  element.elemID.typeName === 'FieldConfigurationScheme' && _.isString(element.value.id)
+const isIssueTypeScreenSchemeInstance = (element: InstanceElement)
+  : element is IssueTypeScreenSchemeInstance =>
+  element.elemID.typeName === 'IssueTypeScreenScheme' && _.isString(element.value.id)
```

Comment:
nit 1 - space between the functions
nit 2 - line breaks in more conventional places (formatting that works when there are multiple arguments as well) + parentheses to make it easier to find the function body

```suggestion
const isFieldConfigurationSchemeInstance = (
  element: InstanceElement,
): element is FieldConfigurationSchemeInstance => (
  element.elemID.typeName === 'FieldConfigurationScheme' && _.isString(element.value.id)
)

const isIssueTypeScreenSchemeInstance = (
  element: InstanceElement,
): element is IssueTypeScreenSchemeInstance => (
  element.elemID.typeName === 'IssueTypeScreenScheme' && _.isString(element.value.id)
)
```

Reply:
v